rfcs/0005-pablo-distribution.md

```diff
@@ -537,92 +635,109 @@ data structures is to be tracked in the staking rewards pallet,
-        pub claimed_shares: Balance,
+    #[derive(RuntimeDebug, PartialEq, Eq, Clone, MaxEncodedLen, Encode, Decode, TypeInfo)]
+    pub enum RewardRatePeriod {
+        PerSecond,
+    }
 
-        /// Pool would stop adding rewards to pool at this block number.
-        pub end_block: BlockNumber,
+    impl RewardRatePeriod {
+        /// Returns the length of the period in seconds.
+        pub fn as_secs(&self) -> NonZeroU64 {
+            match self {
+                RewardRatePeriod::PerSecond => {
+                    sp_std::num::NonZeroU64::new(1).expect("1 is non-zero; qed;")
+                },
+            }
+        }
+    }
 
-        // possible lock config for this pool
-        pub lock: LockConfig<DurationPresets>,
+    /// A reward update states the new reward and reward_rate for a given asset
+    #[derive(RuntimeDebug, Encode, Decode, MaxEncodedLen, Clone, PartialEq, Eq, TypeInfo)]
+    pub struct RewardUpdate<Balance> {
+        /// The rewarding rate that increases the pool `total_reward`
+        /// at a given time.
+        pub reward_rate: RewardRate<Balance>,
     }
 
 Following sections describe the algorithms for various operations on the
 rewards pool based on these data structures.
 
 #### 5.3.3. Staking
 
-<img src="0005-pablo-distribution-assets/images/images/staking.png" width="684" height="873" alt="staking" />
+<img src="images/images/staking.png" width="623" height="873"
+alt="staking" />
 
 #### 5.3.4. Extend Position
 
-<img src="0005-pablo-distribution-assets/images/images/extend-position.png" width="684" height="994" alt="extend position" />
+<img src="images/images/extend-position.png" width="514" height="940"
+alt="extend position" />
 
 #### 5.3.5. Split Position
 
-<img src="0005-pablo-distribution-assets/images/images/split-position.png" width="426" height="344" alt="split position" />
+<img src="images/images/split-position.png" width="371" height="290"
+alt="split position" />
 
-#### 5.3.6. Claim/Unstake
+#### 5.3.6. Unstake
 
-<img src="0005-pablo-distribution-assets/images/images/claim.png" width="500" height="695" alt="claim" />
+<img src="images/images/unstake.png" width="404" height="547"
+alt="unstake" />
 
 #### 5.3.7. Update Reward Pool
 
-<img src="0005-pablo-distribution-assets/images/images/update-reward-pool.png" width="508" height="476" alt="update reward pool" />
+<img src="images/images/update-reward-pool.png" width="441" height="647"
+alt="update reward pool" />
 
 #### 5.3.8. RewardAccumulationHook
 
 Following algorithm should be part of the block hook in the pallet.
 
-<img src="0005-pablo-distribution-assets/images/images/staking-rewards-reward-accumulation-hook.png" width="509" height="501" alt="staking rewards reward accumulation hook" />
+<img src="images/images/staking-rewards-reward-accumulation-hook.png"
+width="451" height="591"
+alt="staking rewards reward accumulation hook" />
+
+#### 5.3.9. Claim
+
+<img src="images/images/claim.png" width="653" height="696"
```

Comment:
The markdown does not render correctly as it is. Need to revert all changes(including the images) other than in this part and Unstake section. Normally what I do is generate the md and then make diff merge between the existing and the generated one to include only the stuff added.

Reply:
Reverted here
231d1ee9b30271600f0c25ac82ef08e009a388db